Add support for GS64 3.7.0

gcotelli opened the pull request.

gcotelli committed:
Move Zinc extensions to its own package

gcotelli committed:
Move Teapot extensions to its own package

gcotelli committed:
Move NeoJSON extensions to its own package

gcotelli committed:
Move JWT extensions to its own package

gcotelli committed:
Move Hyperspace extensions to its own package

gcotelli committed:
Move String extension to its own package

gcotelli committed:
Add GH action to check Core loading in GS64

gcotelli committed:
Fix Core.ston

gcotelli committed:
Remove unnecessary pool dictionary

gcotelli committed:
Do not make direct references to `JWAHMACSHA256`
Use JSONWebToken version in ba-st-dependencies

gcotelli committed:
Use MessageDigestAlgorithm facade

gcotelli committed:
Fix test

gcotelli committed:
Add more components

gcotelli committed:
Add first test package

gcotelli committed:
Fix depdency in component

gcotelli committed:
Add SortFunctions dependency on GS64

gcotelli committed:
Use split: instead of splitOn:

gcotelli committed:
Add application configuration components to rowan specs

gcotelli committed:
Fix Launchpad spec

gcotelli committed:
Move abstract test case

gcotelli committed:
Rename spec

gcotelli committed:
Add application control component

gcotelli committed:
Update dependencies

gcotelli committed:
Add missing package with Hyperspace extensions

gcotelli committed:
Move Pharo-specific application info to its own package

gcotelli committed:
Add application info component

gcotelli committed:
Add GS64 version of Operating System information provider

gcotelli committed:
Add new package

gcotelli committed:
Add GS64 information provider

gcotelli committed:
Merge branch 'gs64_info' into gs64_3.7.0

gcotelli committed:
Make application info tests aware of GS64

gcotelli committed:
Add default providers for GS64

ytsejam78 committed:
Stargate-Loggers package added

ytsejam78 committed:
Bell's LogRecord reference fix

gcotelli committed:
Add HealthCheck component for GS64

gcotelli committed:
Split metrics in Pharo and GS64 packages

gcotelli committed:
Add Metrics component

gcotelli committed:
Remove Smalltalk reference

gcotelli committed:
Fix PrometehousMetricEncoder for GS64

gcotelli committed:
Improve CompositeMetric>>#value

gcotelli committed:
Fix tests

gcotelli committed:
Fix tests

gcotelli committed:
Move Zinc events extensions to its own package

gcotelli committed:
Add HTTP metrics to component

ytsejam78 committed:
Beacon extensions moved to specific package

ytsejam78 committed:
Bell extensions moved to specific package

ytsejam78 committed:
More protocol extended for Bell classes

ytsejam78 committed:
Rowan specs updated for Logger packages

gcotelli committed:
Merge branch 'gs64_3.7.0' into gs64_3.7.0_loggers

gcotelli committed:
Merge pull request #175 from ba-st/gs64_3.7.0_loggers

GS64 Loggers support

gcotelli committed:
Fix loggers tests

gcotelli committed:
Use the textual stack trace dumper by default
Improve naming of stack trace file
Change applicationBaselineName to projectName (breaking backwards compatibility)

gcotelli committed:
Add examples component
Add API-Skeleton to deployment component

gcotelli committed:
Fix references to LaunchpadLogRecord

gcotelli committed:
Access headers using strings and not symbols

gcotelli committed:
Don't use directly instVarNamed:put:

gcotelli committed:
Do not use splitOn:

gcotelli committed:
Fix tests

gcotelli committed:
Fix testFileReferenceToDumpStackTrace on GS64

gcotelli committed:
Add test package to Tests component

gcotelli committed:
Move Teapot extensions to its own package

gcotelli committed:
Fix tests in GS64

gcotelli committed:
Add ZnEntity >> #asString

gcotelli committed:
Add ignore for coverage in GS64 only method

gcotelli committed:
Update docs and remove Pharo 8 and 9 support

gcotelli committed:
Fix markdown lint complaints

jvanecek commented on source/Stargate-Application-Info-Pharo/PharoOperatingSystemInformationProvider.class.st:
```diff
@@ -2,25 +2,25 @@
 "
 Class {
-	#name : #OperatingSystemInformationProvider,
+	#name : #PharoOperatingSystemInformationProvider,
```
Shouldn´t this be UNIXOperatingSystemInformationProvider?

gcotelli replied:
This provider works for Windows too, it only shows less detailed information.

ytsejam78 replied:
I think Juan has a good point. When thinking about an OS we think about Linux, Windows, etc. Maybe a better name is "platform"? I asked Microsoft AI "What is the difference between platform and operating system?" and got:

> In the context of computer systems, the terms **platform** and **operating system** have distinct meanings¹²³.
> 
> An **operating system** (OS) is a software that manages a computer's hardware and software resources and provides common services for computer programs². It acts as an intermediary between applications and hardware, managing how applications access hardware and software resources³. An OS provides a set of interconnected tools such as file systems, interprocess communication, process management, network access, human interface, and shells¹.
> 
> On the other hand, a **platform** is a combination of hardware and software that allows software to run². It is a group of technologies used as a base upon which other applications, processes, or technologies are developed³. For instance, Windows and macOS are two platforms or operating systems¹. When it comes to different versions of Windows like Windows XP and Windows 7, they can be considered different versions of the same platform¹.
> 
> To summarize, an operating system is the software that manages a computer's resources and provides common services for computer programs. A platform is the hardware and software combination that allows software to run².
> 
> Source: Conversation with Bing, 9/28/2023
> (1) Platform vs operating system - Super User. https://superuser.com/questions/514459/platform-vs-operating-system.
> (2) What is the Difference Between Operating System and Platform?. https://similardifferent.com/difference-between-operating-system-and-platform/.
> (3) What is difference between platform and operating system?. https://www.compuhoy.com/what-is-difference-between-platform-and-operating-system/.
> (4) Quick Answer: What is difference between platform and operating system?. https://frameboxxindore.com/other/quick-answer-what-is-difference-between-platform-and-operating-system.html.

ytsejam78 commented on source/BaselineOfStargate/BaselineOfStargate.class.st:
```diff
@@ -21,114 +22,139 @@ BaselineOfStargate >> baseline: spec [
 		project: 'Launchpad-Deployment' copyFrom: 'Launchpad' with: [ spec loads: 'Deployment' ].
 
 	spec
 		baseline: 'ZTimestamp' with: [ spec repository: 'github://svenvc/ztimestamp' ];
```
Does the dependecy with ZTimestamp still exist? Now that stack/fuel dump file name generation includes hhmmss + uuid I think we could avoid loading even in Pharo. Or can be removed without too much effort maybe?

ytsejam78 commented on source/Stargate-API-Skeleton/StargateApplication.class.st:
```diff
@@ -48,12 +65,24 @@ StargateApplication class >> logsDirectory [
+{ #category : #private }
+StargateApplication class >> projectName [
+
+	^ self subclassResponsibility
```
unnecessary return here

gcotelli replied:
It's true, but I use it as an indicator that the implementors are expected to explicitly return something

ytsejam78 commented on source/Stargate-API-Skeleton/StargateApplication.class.st:
```diff
@@ -126,7 +155,7 @@ StargateApplication >> applicationInfoConfiguration [
 StargateApplication >> authAlgorithm [
 
-	^ JWAHMACSHA256 parameterValue
+	^ 'HS256'
```
May not be such a bad idea to write a comment in this method explaining where this string comes from.

ytsejam78 replied:
Same for:
JWTBearerAuthenticationFilterTest >> algorithmName
HTTPBasedRESTfulAPITest >> algorithmName

ytsejam78 commented on source/Stargate-Metrics-Tests/MetricsRESTfulControllerTest.class.st:
```diff
@@ -34,27 +34,26 @@ MetricsRESTfulControllerTest >> testGetMetricsInJson [
-					]
+		equals: resourceController operationalMetricsVersion1dot0dot0MediaType;
+		withJsonFromContentsIn: response do: [ :metrics |
+			self
```
Maybe here and I saw the same as part of: MetricsPluginTest >> testConfiguredRunningAllMetrics include an example of what the metrics look like in each case so that it's not just assert size equals X.

gcotelli replied:
The problem here is that the metric values will change on every test run, the number and kind of metrics are different in every platform and the Prometheus format is not easy to test without controlling all the variables.
testGetMetricsInJson does a bit of what you're asking here by testing the number of metrics available in every platform, and the general structure of each one but without testing the exact values.

gcotelli committed:
Take into account code review comments

- Remove unused ZTimestamp dependency
- Improve authAlgorithm comment cross-referencing the normative docs

gcotelli committed:
Remove dependency

ytsejam78 reviewed (approved):
:clap::clap::clap:
